### DISCUSSION TITLE
Windows ARM64 Support

### DISCUSSION THREAD

**Discussion by TGSAN** [TGSAN/CMWTAT_Digital_Edition#43] [General]
ARM64 support under development, related issues can be discussed here

**Comment by NalCol** [comment]
软件没有任何直观的形式表明哪里卡住了，CPU占用一直很高（15%左右）。
另一款W10 Digital Activation Program也基本是同样情况。
但是 HEU KMS Activator成功以数字激活方式激活系统。
不知道是不是架构问题。

**Comment by TGSAN** [comment]
> 软件没有任何直观的形式表明哪里卡住了，CPU占用一直很高（15%左右）。 另一款W10 Digital Activation Program也基本是同样情况。 但是 HEU KMS Activator成功以数字激活方式激活系统。 不知道是不是架构问题。

目前支持ARM64的版本已经基本开发完了，但是因为代码比较脏还没有提交到GitHub，之前的这个版本用的dll hook的形式去做状态欺骗，理论上是可以利用Windows指令集转换来兼容的，但是事实上跑起来不是很正常。新版本我通过调用未导出符号的方法来计算来避免使用dll hook，事实使用这种上在我的Surface Pro X（ARM64）上可以正常运行，但是主程序还是使用了不支持arm 的旧版本 .NET Framework，在不支持x64转义的Windows 10上面不知道能不能正常运行。目前还在解决这部分问题，解决之后应该就可以发布了。

**Comment by TGSAN** [comment]
如果急着激活的话可以临时尝试先从x86或amd64的设备上使用 CMWTAT 激活，然后在arm64设备上使用激活疑难解答，通过Microsoft账号的移进行许可迁移（走官方的正版许可自助证迁），然后在x86或amd64的设备上重新使用 CMWTAT 激活。需要注意的是两台设备的Windows SKU版本需要一致（可以使用 CMWTAT 的版本转换）。

**Comment by TGSAN** [comment]
#42 更新增加适用于 ARM64 的实验性支持

**Reply by NalCol** [reply]
好！👏👏

**Comment by Kin-Zhang** [comment]
刚刚用的arm window10安装的 选择core 激活成功了～

**Comment by NalCol** [comment]
![image](https://github.com/TGSAN/CMWTAT_Digital_Edition/assets/58576200/607d7a89-272b-4286-acc5-5b7d3970fa4e)
@TGSAN 在M$Store抓到你^_^

**Reply by TGSAN** [reply]
确实是我（

**Reply by NalCol** [reply]
AMP做的确实不够好，经常出bug，我表示网页端除了无损大抵是好很多罢（
